app/javascript/components/workflows/student/exams/show/types.ts

```diff
@@ -439,9 +486,24 @@ export interface MatchingState {
   [index: number]: number;
 }
 
+export interface MatchingPromptWithAnswer {
+  html: HTMLVal;
+  answer: number;
+}
+
+export interface MatchingInfoWithAnswer extends Omit<MatchingInfo, 'prompts'> {
+  prompts: MatchingPromptWithAnswer[];
+}
+
 export type BodyItem =
   HTMLVal | AllThatApplyInfo | CodeInfo | YesNoInfo |
-  CodeTagInfo | MultipleChoiceInfo | TextInfo | MatchingInfo;
+  CodeTagInfo | MultipleChoiceInfo |
+  TextInfo | MatchingInfo;
+
+export type BodyItemWithAnswer =
+  HTMLValWithAnswer | AllThatApplyInfoWithAnswer | CodeInfoWithAnswer | YesNoInfoWithAnswer |
```

Comment:
How can an HTMLVal have an answer?

Reply:
It needs to have a `{ NO_ANS: true }` for rails to not throw it away.

The type is `answer: NoAnswerState`